src/core/sort.cc

```diff
@@ -756,15 +786,15 @@ class SortContext {
       dt::parallel_for_static(n,
         [&](size_t j) {
           TI t = xi[o[j]];
-          xo[j] = t == una? 0 :
+          xo[j] = t == una? replace_una :
```

Comment:
Previously we had a map
```
NA -> 0
min -> 1
...
max -> (max-min + 1)
```
However, right now when NA is in last position the mapping is
```
min -> 1
...
max -> (max-min + 1)
NA -> (max-min + 2)
```
which can potentially overflow for a sufficiently wide range min..max. For example, try sorting an array that contains both `-MAX`, `MAX` and `NA`, where `MAX=2**32-1`.

Reply:
Good point, Wouldn't overflow happen when we map `max -> (max-min+1)` as well?

Just experimented with the following
```python
MAX = pow(2,32)-1
dt_int = dt.Frame(x=[0,1,None,MAX,None,-MAX,None])
dt_int[:, :, dt.sort(0, reverse=False, na_position="last")]
```
and got 
```code
  |           x
-- + -----------
 0 |  4294967295
 1 |           1
 2 |           0
 3 | -4294967297
 4 |          NA
 5 |          NA
 6 |          NA
```
Obviously, the column.stypes changed from `int32` to `int64` and It also happens when we map `max -> (max-min+1)` and `NA -> 0`

Reply:
Ah, my bad. The largest value that can fit into int32 is `MAX = 2**31 - 1`. With this value, I'm getting (with your branch)
```
>>> dt_int[:, :, dt.sort(0, reverse=False, na_position="last")]
   |           x
-- + -----------
 0 |          NA
 1 |          NA
 2 |          NA
 3 | -2147483647
 4 |           0
 5 |           1
 6 |  2147483647

[7 rows x 1 column]
```

Reply:
Yeah. So `Max` ends up getting mapped to `std::numeric_limits<unsigned int>::max()`. Anything above that is an overflow. do you have any suggestions? :worried:

Reply:
Well, the formula that is computed here is basically
```
xo[j] = (x == NA)? 0 : 
          ASC? (x - min + 1) :
               (max - x + 1);
```
That "+1" makes it so that `x==min` maps into the value 1, which is right after the map value of NA (which is 0). But with the alternative placement of NAs, you'd have to modify this formula so that it says
```
xo[j] = (x == NA)? (max - min + 1) : 
          ASC? (x - min) :
               (max - x);
```
So you can either make a bigger nested if-condition, or maybe try to combine both cases into one. For example, write `(x - min + NAFIRST)` where `NAFIRST=(na_pos=='first')?1:0`.